src/store.c

```diff
@@ -93,6 +93,28 @@ static int pid_cmdline_strscpy(int procfd, char *buffer, size_t buf_len, int pid
         return 0;
 }
 
+static void garbage_collect_dead_processes(struct ps_struct *ps_first) {
+        struct ps_struct *ps;
+        struct ps_struct *ps_next;
+
+        ps = ps_first;
+        while ((ps_next = ps->next_running)) {
+                if (!ps_next->still_running) {
+                        /* close the stream and fds */
+                        ps_next->schedstat = safe_close(ps_next->schedstat);
```

Comment:
These assignments aren't necessary. I'm just did this for consistency with how it's done in [bootchart.c](https://github.com/systemd/systemd-bootchart/blob/master/src/bootchart.c#L468).